python/README.md

```diff
@@ -35,7 +35,7 @@ Development is made easy using [Poetry](https://python-poetry.org/docs/#installa
 
 ```bash
 poetry shell
-poetry install -E pyarrow
+poetry install -E pyarrow -E hive -E s3fs
```

Comment:
tests don't pass without `s3fs` and so I figured we can align this with the make file, which uses `poetry install -E pyarrow -E hive -E s3fs`

Reply:
Thanks @nastra, great catch! It seems that this is outdated. What do you think of:
```sh
make install
make test
```
This way we just have to maintain the `Makefile`, that is also being used in the CI.
Makefile: https://github.com/apache/iceberg/blob/master/python/Makefile#L18-L20

Reply:
makes sense, updated